Readded device_assert skipping in index and index_put (and also added

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112093

copy to noop pass)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D50821678](https://our.internmc.facebook.com/intern/diff/D50821678)

Chillee committed:
Readded device_assert skipping in index and index_put (and also added
copy to noop pass)

[ghstack-poisoned]

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

lezcano commented on torch/_inductor/lowering.py:
```diff
@@ -2915,6 +2917,7 @@ def index_put_impl_(self, indices, values, accumulate, check):
         indices_loaders,
         indexed_size,
         None,
+        add_asserts=check,
```
Either change all the `check` parameters to `add_asserts` (a better name IMO) or name this one `check` as well.

lezcano reviewed (approved):
The device_assert change LGTM modulo a notation nit. The other changes seem unrelated so they may belong to a different PR?

Chillee commented:
@lezcano I'm uhh... smuggling two changes in here. The other change is pretty minor though.

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge

Chillee commented:
@pytorchbot merge

izaitsevfb commented:
@pytorchbot revert -m "Sorry, trying to resolve a conflict with intern, and unblock the revert of #108690" -c ghfirst

Chillee commented:
@izaitsevfb lmk when I can reland.

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge

izaitsevfb commented:
> @izaitsevfb lmk when I can reland.

After we either revert #108690 or re-land it internally (it fails some tests, but at this point that might be a better option).

Chillee commented:
@pytorchbot cancel

Chillee commented:
@pytorchbot merge

ZainRizvi commented:
@pytorchbot revert -m "Stack breaks internal builds" -c ghfirst

malfet commented:
> @pytorchbot revert -m "Stack breaks internal builds" -c ghfirst

Looks like it breaks some fbgemm_gpu and torchbench test cases.

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee commented:
@chillee has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50821678).

Chillee committed:
Update on "Readded device_assert skipping in index and index_put (and also added"


copy to noop pass)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D50821678](https://our.internmc.facebook.com/intern/diff/D50821678)

[ghstack-poisoned]

Chillee commented:
Landed in https://github.com/pytorch/pytorch/pull/112476
